pkg/controller/jobs/pod/pod_multikueue_adapter.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/features"
+	"sigs.k8s.io/kueue/pkg/util/api"
+	clientutil "sigs.k8s.io/kueue/pkg/util/client"
+)
+
+type multikueueAdapter struct{}
+
+var _ jobframework.MultiKueueAdapter = (*multikueueAdapter)(nil)
+
+func (b *multikueueAdapter) SyncJob(ctx context.Context, localClient client.Client, remoteClient client.Client, key types.NamespacedName, workloadName, origin string) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	localPod := corev1.Pod{}
+	err := localClient.Get(ctx, key, &localPod)
+	if err != nil {
+		return err
+	}
+
+	remotePod := corev1.Pod{}
+	err = remoteClient.Get(ctx, key, &remotePod)
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	// the remote pod exists
+	if err == nil {
+		if FromObject(&localPod).IsSuspended() {
+			// Ensure the pod is unsuspended before updating its status; otherwise, it will fail when patching the spec.
+			log.V(2).Info("Skipping the sync since the local pod is still suspended")
+			return nil
+		}
+
+		remoteFinished := false
+		if remotePod.Status.Phase == corev1.PodSucceeded || remotePod.Status.Phase == corev1.PodFailed {
```

Comment:
Probably [Unknown](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) too? That is also an indication that we don't know what's happening, and is better to consider it to be equal to Fail rather than leave it on pending.

Reply:
Yeah that is a good idea wasn't sure if that was what we wanted but I suppose it is better to side with caution in cases like this :+1: